### DISCUSSION TITLE
Are factories still producing a single product in SCML 2021

### DISCUSSION THREAD

**Discussion by umutcakan** [yasserfarouk/scml#29] [collusion-qa] [answered]
Hey there,

Are factories going to produce multiple products or the my_input_products and my_input_product does not differ in the SCML 2021?

my_input_product: The input product to the factory controlled by the agent.
my_input_products: All input products of a factory controlled by the agent. Currently, it is always a list of one item. For future compatibility.

Best regards.

**Comment by yasserfarouk** [comment, marked as answer]
They produce a single output. the plural forms are for future proofing of the interface. 
the only difference is that my-input-product Is an integer (say 2) but my-input-products is a list of one integer (say [2]).

I modified the question title to make it more helpful for others. I hope you do not object.

**Comment by umutcakan** [comment]
Thanks for the answer Yasser! Title was not informative, thanks for fixing it too ! :)
